wormhole-connect/src/hooks/useFetchTransactionHistory.ts

```diff
@@ -0,0 +1,243 @@
+import { useSelector } from 'react-redux';
+import { amount as sdkAmount, chainIdToChain } from '@wormhole-foundation/sdk';
+
+import config from 'config';
+import { getTokenById, getWrappedToken } from 'utils';
+
+import type { RootState } from 'store';
+import type { Chain } from '@wormhole-foundation/sdk';
+import type { RelayerFee } from 'store/relay';
+
+export interface Transaction {
+  // Transaction hash
+  txHash: string;
+
+  // Stringified addresses
+  sender?: string;
+  recipient: string;
+
+  amount: string;
+  amountUsd: number;
+
+  toChain: Chain;
+  fromChain: Chain;
+
+  // Source token address
+  tokenAddress: string;
+  tokenKey: string;
+  tokenDecimals?: number;
+
+  // Destination token
+  receivedTokenKey?: string;
+  receiveAmount?: string;
+  relayerFee?: RelayerFee;
+
+  // Amount of native gas being received, in destination gas token units
+  // For example 1.0 is 1.0 ETH, not 1 wei
+  receiveNativeAmount?: number;
+
+  // Timestamps
+  senderTimestamp?: string;
+  receiverTimestamp?: string;
+}
+
+type Props = {
+  page?: number;
+  pageSize?: number;
+};
+
+const useFetchTransactionHistory = (
+  props: Props,
+): {
+  transactions: Array<Transaction>;
+  error: string;
+  isFetching: boolean;
+  hasMore: boolean;
+} => {
+  const [transactions, setTransactions] = useState<Array<Transaction>>([]);
+  const [error, setError] = useState('');
+  const [isFetching, setIsFetching] = useState(false);
+  const [hasMore, setHasMore] = useState(true);
+
+  const { sending: sendingWallet } = useSelector(
+    (state: RootState) => state.wallet,
+  );
+
+  const { page = 0, pageSize = 30 } = props;
+
+  const parseTokenBridgeTx = (tx) => {
+    const { content = {}, data = {}, sourceChain = {}, targetChain = {} } = tx;
+    const { standarizedProperties = {} } = content;
+
+    const fromChainId = standarizedProperties.fromChain || sourceChain?.chainId;
+    const toChainId = standarizedProperties.toChain || targetChain?.chainId;
+
+    const fromChain = chainIdToChain(fromChainId);
+
+    // Skip if we don't have the source chain
+    if (!fromChain) {
+      return;
+    }
+
+    const tokenConfig = getTokenById({
+      chain: fromChain,
+      address: standarizedProperties.tokenAddress,
+    });
+
+    // Skip if we don't have the source token
+    if (!tokenConfig) {
+      return;
+    }
+
+    const toChain = chainIdToChain(toChainId) as Chain;
+    const toChainConfig = config.chains[toChain];
+
+    const receivedTokenKey =
+      tokenConfig.nativeChain === toChain
+        ? tokenConfig.key
+        : getWrappedToken(tokenConfig)?.key;
+
+    // Reverse engineering the conversion rate:
+    //   data.tokenAmount is the actual value and standarizedProperties.amount is the denomination.
+    //   There has been unexpected conversaion rates in API responses. Therefore we are
+    //   sniffing out the decimals from the conversion rate of between these two values.
+    const conversionRate =
```

Comment:
I'm not 100% sure if this is a good way to get the decimals. TBH the data API returns changes from 6 to 9 decimals. What would be the best way to calculate these decimals for a chain/token?

Reply:
We have `getTokenDecimals` in `utils` which can give you the decimals for a given token. This is just based on which chain it's on. Does this work?

```ts
const tokenId = {
  chain: fromChain,
  address: standarizedProperties.tokenAddress,
};
const tokenConfig = getTokenById(tokenId);
const tokenDecimals = getTokenDecimals(tokenId);
const { amount, fee } = standardizedProperties;
const receiveAmount = BigInt(amount) - BigInt(fee);
const receiveAmountDisplay = sdkAmount.display({
  amount: amountOutBaseUnits.toString(),
  decimals: tokenDecimals
});
```

We should use BigInts for arithmetic whenever possible

Reply:
@artursapek that gives me the decimals in token config, which are usually larger than the actual. I can use bigInt type, but the second value is decimal, so it will be a bit more tricky. Let me give it a try.

Reply:
@artursapek I think this one seems to be working:
```
const decimals = Math.min(
      8,
      getTokenDecimals(
        fromChain,
        tokenConfig.nativeChain === fromChain ? 'native' : tokenConfig.tokenId,
      ),
    );
```

Reply:
OK. FYI, the string `'native'` is meant to represent the native gas token on a chain. I don't think you are using it correctly here; `tokenConfig.nativeChain` just represents which chain the token is native to; it doesn't indicate that the token is the _native gas token_. It's kind of confusing.

Maybe a dumb question, but is it possible Wormholescan's API just always uses decimals of `8`?

Reply:
Also, the comment about BigInt is that it's just better to use BigInt when subtracting the fee from the original amount, to get the receive amount. We're working with base units which are integers so we should avoid using the default `Number`, which is a floating point, to do arithmetic with base units.

Reply:
It might be just 8 for all decimals from the API. We can confirm this on Monday I believe 👍

Reply:
I've als pushed a change to use BigInt for operations.